### PR TITLE
Api routes

### PR DESCRIPTION
**Pull request by MarkPThomas**
Just setting up a basic skeleton structure of expected API calls, connecting them with a higher level set of methods that delegate to the matching lower level methods for each database.

Right now I am mirroring the current API state but will add improved methods later. While the API might be improved, for backwards compatibility we should do this so that the user has no idea anything has changed on the back end.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
+const modelsSecondary = require('./db/secondary');
+
+// Initialize server w/ DB
+sequelize.sync({ force: eraseDatabaseOnSync }).then( () => {
+  if (eraseDatabaseOnSync) {
+    seedDatabase(models);
+  }
+
+  app.listen(port, () => {
+    console.log(`NSA is listening in at http://localhost:${port}`);
```
Comment (aleksandar-cakic) [inline server/index.js:20]: Nice! I use 'Port x is on fire!!' lol

**Review by aleksandar-cakic — APPROVED** [review]
Looks great! I added a few comments/questions

File: server/db/primary/models/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      console.log('getReviewsByProduct error:', error);
+      reject(error);
+    } else {
+      resolve(results);
+    }
+  });
+}
+module.exports.getReviewsByProduct = getReviewsByProduct;
+
+const getReview = (reviewId) => {
```
[the remark this answers is not in the document]
Comment (MarkPThomas) [reply server/db/primary/models/index.js:23]: I figure each review has a unique id, which is separate from the product id. So you can always get a review directly with that ID. In other cases you get all reviews associated with a given product. My entity diagram should make this more clear.

File: server/db/primary/index.js [hunk 3]
```diff
@@ -1,21 +1,49 @@
...
-db.connectAsync()
-  .then(() => console.log(`Connected to ${database} database as ID ${db.threadId}`))
-  .then(() => db.queryAsync(`CREATE DATABASE IF NOT EXISTS ${database}`))
-  .then(() => db.queryAsync(`USE ${database}`))
-  .then(() => createTables(db));
+Object.keys(models).forEach(key => {
+  if ('associate' in models[key]) {
+    models[key].associate(models);
+  }
+});
```
[the remark this answers is not in the document]
Comment (MarkPThomas) [reply server/db/primary/index.js:47]: This was in an example for using postgre with sequelize. I'll need to look more into it, but my general impression was that it was linking tables together. Or may be irrelevant or stripped out. I'll get to it as I construct the database.

**Pull request merged**